src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -2758,4 +2758,463 @@ .maxstack  2
+            """;
+        CompileAndVerify(source, expectedOutput: "C111").VerifyDiagnostics(
+            // (7,21): warning CS9508: Modifier of parameter 'ref readonly int x' doesn't match the corresponding parameter 'in int x' in hidden member.
+            //     public new void M(ref readonly int x) => System.Console.Write("C" + x);
+            Diagnostic(ErrorCode.WRN_HidingDifferentRefness, "M").WithArguments("ref readonly int x", "in int x").WithLocation(7, 21));
+    }
+
+    [Fact]
+    public void Hiding_RefReadonly_In()
+    {
+        var source = """
+            class B
+            {
+                public void M(ref readonly int x) => System.Console.Write("B" + x);
+            }
+            class C : B
+            {
+                public void M(in int x) => System.Console.Write("C" + x);
+                static void Main()
+                {
+                    var x = 111;
+                    new C().M(in x);
+                }
+            }
+            """;
+        CompileAndVerify(source, expectedOutput: "C111").VerifyDiagnostics(
+            // (7,17): warning CS0108: 'C.M(in int)' hides inherited member 'B.M(ref readonly int)'. Use the new keyword if hiding was intended.
+            //     public void M(in int x) { }
+            Diagnostic(ErrorCode.WRN_NewRequired, "M").WithArguments("C.M(in int)", "B.M(ref readonly int)").WithLocation(7, 17));
+    }
+
+    [Fact]
+    public void Hiding_RefReadonly_In_New()
+    {
+        var source = """
+            class B
+            {
+                public void M(ref readonly int x) => System.Console.Write("B" + x);
+            }
+            class C : B
+            {
+                public new void M(in int x) => System.Console.Write("C" + x);
+                static void Main()
+                {
+                    var x = 111;
+                    new C().M(in x);
+                }
+            }
+            """;
+        CompileAndVerify(source, expectedOutput: "C111").VerifyDiagnostics(
+            // (7,21): warning CS9508: Modifier of parameter 'in int x' doesn't match the corresponding parameter 'ref readonly int x' in hidden member.
+            //     public new void M(in int x) => System.Console.Write("C" + x);
+            Diagnostic(ErrorCode.WRN_HidingDifferentRefness, "M").WithArguments("in int x", "ref readonly int x").WithLocation(7, 21));
+    }
+
+    [Theory, CombinatorialData]
+    public void Hiding_RefReadonly_NotIn([CombinatorialValues("ref", "out")] string modifier)
+    {
+        var source = $$"""
+            class B
+            {
+                public void M1({{modifier}} int x) => throw null!;
+                public void M2(ref readonly int x) { }
+            }
+            class C : B
+            {
+                public void M1(ref readonly int x) { }
+                public void M2({{modifier}} int x) => throw null!;
+            }
+            """;
+        CreateCompilation(source).VerifyDiagnostics();
```

Comment:
>CreateCompilation(source).VerifyDiagnostics(); [](http://example.com/codeflow?start=8&length=46)

Is there a way to consume base members through instance of ```C```?

Reply:
Yes with the right modifiers, except when base is `ref` and hiding is `ref readonly`, there it's possible using casting. Will extend the test to demonstrate.